fixtures/fixtures.js

```diff
@@ -90,12 +102,27 @@ Fixtures.apply = function (mock) {
     mock.onGet(/\/traders\/2\/vouchers/).reply(200, this.traderVouchers["2"]);
 
     mock.onPost('/vouchers').reply(function (request) {
-        // returns a success regardless!
-        // TODO : better emulation of server side validated responses;
+        const requestPayload = JSON.parse(request.data);
+
+        const success = [];
+        const fail = [];
+        const invalid = [];
+
+        requestPayload.vouchers.forEach((voucherNumber) => {
```

Comment:
Agreed - fixtures represent a given app-state. You're writing a loose emulation of the serverside in here, which is  more than we need (even though it is more flexible).